### PR TITLE
[ADD] account_move_locking V7

### PR DESCRIPTION
**Pull request by vrenaville**
It's a backport a account_move_locking from V8

It's allow to lock a move to prevent it to be modified, even if you have account_cancel installed

### CODE REVIEW

File: account_move_locking/README.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+`here <https://github.com/OCA/account-financial-tools/issues/new?body=module:%20account_move_locking%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
+
+Credits
+=======
+
+Contributors
+------------
+
+* Vincent renaville <vincen.renaville@camptocamp.com>
```
Comment (mdietrichc2c) [inline account_move_locking/README.rst]: Correct mail/name syntax

File: account_move_locking/README.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+This module add the ability to lock move for modification 
+
+Configuration
+=============
+
+
+
+Usage
+=====
+In order the locked the move you need to follow this process
```
Comment (mdietrichc2c) [inline account_move_locking/README.rst]: `In order to lock the move, you need to follow this process:`

File: account_move_locking/README.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    * You need to post your move, with the standard wizard Post Journal Entries
+      (Invoicing -> Periodic Processing -> Draft Entries -> Post Journal Entries) 
+    * After the you can use the wizard Lock Journal Entries
```
Comment (mdietrichc2c) [inline account_move_locking/README.rst]: Missing word?

File: account_move_locking/i18n/fr.po [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+#, python-format
+msgid "Lock Account Move"
+msgstr "Verrouiller mouvement comptable"
+
+#. module: account_move_locking
+#: model:ir.actions.act_window,name:account_move_locking.action_lock_account_move
+#: model:ir.ui.menu,name:account_move_locking.menu_lock_account_moves
+#: view:lock.account.move:0
+msgid "Lock Journal Entries"
+msgstr "Verrouiller les écritues comptables"
```
Comment (mdietrichc2c) [inline account_move_locking/i18n/fr.po]: écritures

File: account_move_locking/i18n/fr.po [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+#. module: account_move_locking
+#: field:lock.account.move,period_ids:0
+msgid "Period"
+msgstr "Période"
+
+#. module: account_move_locking
+#: code:addons/account_move_locking/wizard/account_lock_account_move.py:59
+#, python-format
+msgid "Unposted move in period/jounal                    selected, please post it before                    locking them"
+msgstr "Ecritures non postées dans la sélection journal/période, Il faut poster les écritures avant de les verrouiller"
```
Comment (mdietrichc2c) [inline account_move_locking/i18n/fr.po]: s/Il/il (no caps).

File: account_move_locking/wizard/account_lock_account_move.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+from tools.translate import _
+
+
+class lock_account_move(orm.TransientModel):
+    _name = "lock.account.move"
+    _description = "Lock Account Move"
+
+    _columns = {
+        'journal_ids': fields.many2many('account.journal',
+
```
Comment (mdietrichc2c) [inline account_move_locking/wizard/account_lock_account_move.py]: Unnecessary line break

File: account_move_locking/wizard/account_lock_account_move.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        'period_ids': fields.many2many('account.period',
+                                       rel='wizard_lock_account_move_period',
+                                       string='Period',
+                                       required=True,
+                                       domain=[('state', '<>', 'done')])
+    }
+
+    def lock_move(self, cr, uid, ids, context=None):
+        obj_move = self.pool['account.move']
+        data = self.browse(cr, uid, ids, context=context)[0]
```
Comment (mdietrichc2c) [inline account_move_locking/wizard/account_lock_account_move.py:49]: Better to check for len(ids) > 1 and put it as a parameter

File: account_move_locking/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+##############################################################################
+{
+    "name": "Move locked to prevent modification",
+    "version": "1.0",
```
Comment (guewen) [inline account_move_locking/__openerp__.py:22]: `description` is required in 7.0

File: account_move_locking/wizard/account_lock_account_move.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        draft_move_ids = obj_move.search(cr, uid,
+                                         [('state', '=', 'draft'),
+                                          ('journal_id',
+                                           'in',
+                                           journal_ids),
+                                          ('period_id', 'in',
+                                           period_ids)],
+                                         order='date',
+                                         context=context)
+        if draft_move_ids:
```
Comment (mdietrichc2c) [inline account_move_locking/wizard/account_lock_account_move.py:59]: Line break before cr, uid, ... (allows more space for each search criteria)

File: account_move_locking/README.rst [hunk 10]
```diff
@@ -0,0 +1,52 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Account move locking
+====================
+
+This module add the ability to lock move for modification
```
Comment (nbessi) [inline account_move_locking/README.rst:7]: This module add the ability to lock and prevent any modification of given financial moves

Something like that would be better no ?

File: account_move_locking/account.py [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+    _columns = {
+        'locked': fields.boolean('Locked', readonly=True),
+    }
+
+    def write(self, cr, uid, ids, vals, context=None):
+        for move in self.browse(cr, uid, ids, context=context):
+            if move.locked:
+                raise orm.except_orm(
+                    _(u'Move Locked!'), move.name)
```
Comment (nbessi) [inline account_move_locking/account.py:36]: weird indent but ok also we may want to extract following code into a function to dry a little bit the code

```
        for move in self.browse(cr, uid, ids, context=context):
            if move.locked:
                raise orm.except_orm(
                    _(u'Move Locked!'), move.name
```

File: account_move_locking/wizard/account_lock_account_move.py [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    def lock_move(self, cr, uid, ids, context=None):
+        obj_move = self.pool['account.move']
+        data = self.browse(cr, uid, ids, context=context)[0]
+        journal_ids = [journal.id for journal in data.journal_ids]
+        period_ids = [period.id for period in data.period_ids]
+        draft_move_ids = obj_move.search(
+            cr, uid, [('state', '=', 'draft'),
+                      ('journal_id', 'in', journal_ids),
+                      ('period_id', 'in', period_ids)],
+            order='date',
```
Comment (nbessi) [inline account_move_locking/wizard/account_lock_account_move.py:56]: I will add a `limit=1`

**Comment by nbessi** [comment]
Some details to fix, but seems globally OK. It would be nice to forward port those fixes.
Also is it normal to not be able to unlock moves from UI? 

If it is deliberate, a message in wizard form that warn he end user that is a definitve action should be added.

**Comment by gurneyalex** [comment]
@vrenaville I just fixed the 7.0 runbot build in #287 which was merged. Can you rebase this PR over origin/7.0?

**Comment by gurneyalex** [comment]
travis failure seems unrelated (and 7.0 is red :angry:)

**Comment by pedrobaeza** [comment]
@gurneyalex is this still interesting for you?

**Pull request closed without merging**